src/translate.rs

```diff
@@ -1081,56 +1173,113 @@ impl FromGlib<glib_sys::gboolean> for bool {
     }
 }
 
-impl FromGlib<u32> for char {
-    #[inline]
-    fn from_glib(val: u32) -> char {
-        char::from_u32(val).expect("Valid Unicode character expected")
-    }
-}
-
 impl FromGlib<i32> for Ordering {
     #[inline]
     fn from_glib(val: i32) -> Ordering {
         val.cmp(&0)
     }
 }
 
-impl FromGlib<u32> for Option<char> {
+/// Translate from a Glib type which can result in an undefined and/or invalid value.
+pub trait TryFromGlib<G>: Sized {
+    type Error;
+    fn try_from_glib(val: G) -> Result<Self, Self::Error>;
+}
+
+/// Error type for `TryFromGlib` when the Glib value is None.
+#[derive(Debug, PartialEq, Eq)]
+pub struct NoneError;
```

Comment:
These generic error names might be problematic with code using `use glib::translate::*` as exists quite a lot. There are `NoneError` also elsewhere.

But maybe this is not a problem in practice.

Reply:
I tried prefixing with `TryFromGlib`. See https://github.com/fengalin/gtk-rs/commit/c47eae5a1d1910c30788343fc7c718e7fdb3cf69

Reply:
Maybe a bit verbose. `Translate` would be shorter, but less informative.

Reply:
Or we just leave it unprefixed until it actually becomes a problem

Reply:
Hum... I think we should stick with a name now.

Reply:
See above in the initial comment "But maybe this is not a problem in practice." :) If all the names seem ugly, let's just keep it and worry about conflicts if they actually happen. Seems very unlikely in practice.

Reply:
I saw your comment :) But, I also agree it would be better to use a prefix. I also think it's the usual way of naming errors. Here are my proposals (taking the longest affected `struct`):

- `NoneOrInvalidError`
- `TryFromGlibNoneOrInvalidError`
- `TranslateNoneOrInvalidError`
- `GlibNoneOrInvalidError`

Maybe the last one :) What do you think? Also @GuillaumeGomez.

Reply:
I would vote for the last one too, @GuillaumeGomez ?

Reply:
No preference so let's go for the last one. :)